q2cli/handlers.py

```diff
@@ -267,9 +268,16 @@ def get_type(self):
         return mapping[self.ast['name']]
 
     def get_click_options(self):
-        yield click.Option(['--' + self.cli_name],
-                           type=self.get_type())  # Use the ugly lookup above
+        type = self.get_type()  # Use the ugly lookup above
+        if type is bool:
+            no_name = self.prefix + 'no_' + self.name
+            cli_no_name = q2cli.util.to_cli_name(no_name)
+            yield click.Option(['--' + self.cli_name + '/--' + cli_no_name])
+        else:
+            yield click.Option(['--' + self.cli_name], type=type)
 
     def get_value(self, arguments, fallback=None):
         value = self._locate_value(arguments, fallback)
+        if self.get_type() is bool:
+            value = self.semtype.encode(value)
```

Comment:
I would probably just do an early `return value` here instead of encoding to decode.